src/index.js

```diff
@@ -80,8 +80,12 @@ export async function run() {
 
     let newSessionExtra = ""
     if (core.getInput("limit-access-to-actor") === "true") {
-      const { actor } = github.context
-      const octokit = new Octokit()
+      const { actor, apiUrl } = github.context
+
+      const octokit = new Octokit({
+        baseUrl: apiUrl,
+        auth: process.env.GITHUB_TOKEN
```

Comment:
Great! This also addresses the infrequent problem I encountered recently where rate limiting of GitHub hosted build agents kicked in due to _other_ workflows issuing unauthenticated API requests.

Reply:
(I think by default GitHub Actions does not pass the GITHUB_TOKEN into the GitHub Action - the user would have to manually pass it over)

Reply:
> (I think by default GitHub Actions does not pass the GITHUB_TOKEN into the GitHub Action - the user would have to manually pass it over)

I think you are right. But we can use [this trick](https://github.com/git-for-windows/setup-git-for-windows-sdk/pull/552/commits/7f964dbf688b47f9b72d5a1a54dfad2adeea7036) instead of requiring the user to manually specify it.

I'll whip up a PR.

Reply:
cool we should also wire it up with brew I think: https://github.com/mxschmitt/action-tmate/issues/69#issuecomment-1369973140

Reply:
> cool we should also wire it up with brew I think: [#69 (comment)](https://github.com/mxschmitt/action-tmate/issues/69#issuecomment-1369973140)

Good point. I opened https://github.com/mxschmitt/action-tmate/pull/150 and will augment it accordingly.

Reply:
@mxschmitt [augmented](https://github.com/mxschmitt/action-tmate/pull/150/commits/5d54988c620136f655e2d5a4dfa94ce59709a78e).

Reply:
So I tested this on my teams GHES and the token was available via the `process.env` call . I had originally implemented this via passing the token and then realized it could be simpler.  

According to the docs github context has the token: 
https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

So at the least I think you could do this:
```
...
with:
  github-token: ${{ github.token }}
```

Reply:
Okay I see we added: 
```
    default: ${{ github.token }}
```
Nice! That's even better.  It's great when the user doesn't even have to be aware of the difference in where they're running the action.